spec/rspec/support/object_inspector_spec.rb

```diff
@@ -92,6 +92,46 @@ def with_date_loaded
             matcher_without_a_description.inspect)
         end
       end
+
+      context "with double objects" do
+        it "displays double name" do
+          double = double('Foo')
+          allow(double).to receive(:name).and_return("Foo")
+          ObjectInspector.register double.class do |double|
+            "Double (#{double.name})"
+          end
+          expect(ObjectInspector.inspect(double)).to eq("Double (Foo)")
+        end
+      end
+
+      context "with arbitary type" do
+        class User
+          attr_accessor :name
+
+          def initialize(name)
+            @name = name
+          end
+        end
+
+        it "works with custom class" do
+          _user = User.new("FooBar")
+          ObjectInspector.register User do |user|
+            "User (#{user.name})"
+          end
+          expect(ObjectInspector.inspect(_user)).to eq("User (FooBar)")
+        end
+
+        it "works with subclasses" do
+          class Admin < User
+          end
+
+          _admin = Admin.new("FooBar")
+          ObjectInspector.register Admin do |admin|
+            "Admin (#{admin.name})"
+          end
+          expect(ObjectInspector.inspect(_admin)).to eq("Admin (FooBar)")
+        end
```

Comment:
@myronmarston What I understood is, expected output of `ObjectInspector.inspect(_admin)` should be `User (FooBar)`. Am I correct?